cartography/intel/okta/oktaintel.py

```diff
@@ -0,0 +1,1193 @@
+    :return: Nothing
+    """
+
+    ingest_statement = """
+    MATCH (org:OktaOrganization{id: {ORG_ID}})
+    WITH org
+    UNWIND {USER_LIST} as user_data
+    MERGE (new_user:OktaUser{id: user_data.id})
+    ON CREATE SET new_user.firstseen = timestamp()
+    SET new_user.first_name = user_data.first_name,
+    new_user.last_name = user_data.last_name,
+    new_user.login = user_data.login,
+    new_user.email = user_data.email,
+    new_user.second_email = user_data.second_email,
+    new_user.mobile_phone = user_data.mobile_phone,
+    new_user.created = user_data.created,
+    new_user.activated = user_data.activated,
+    new_user.status_changed = user_data.status_changed,
+    new_user.last_login = user_data.last_login,
+    new_user.okta_last_updated = user_data.okta_last_updated,
+    new_user.password_changed = user_data.password_changed,
+    new_user.transition_to_status = user_data.transition_to_status,
+    new_user.lastupdated = {okta_update_tag}
+    WITH new_user, org
+    MERGE (org)-[org_r:RESOURCE]->(new_user)
+    ON CREATE SET org_r.firstseen = timestamp()
+    SET org_r.lastupdated = {okta_update_tag}
+    WITH new_user
+    MERGE (h:Human{email: new_user.email})
+    ON CREATE SET new_user.firstseen = timestamp()
+    SET h.lastupdated = {okta_update_tag}
+    MERGE (h)-[r:IDENTITY_OKTA]->(new_user)
+    ON CREATE SET new_user.firstseen = timestamp()
+    SET h.lastupdated = {okta_update_tag}
+    """
+
+    neo4j_session.run(
+        ingest_statement,
+        ORG_ID=okta_org_id,
+        USER_LIST=user_list,
+        okta_update_tag=okta_update_tag,
+    )
+
+
+def _sync_okta_users(neo4j_session, okta_org_id, okta_update_tag):
+    """
+    Sync okta users
+    :param neo4j_session: Session with Neo4j server
+    :param okta_org_id: Okta organization id to sync
+    :param okta_update_tag: The timestamp value to set our new Neo4j resources with
+    :return: Nothing
+    """
+
+    logger.debug("Syncing Okta users")
+    user_client = _create_user_client(okta_org_id)
+    data = _get_okta_users(user_client)
+
+    _load_okta_users(neo4j_session, okta_org_id, data, okta_update_tag)
+
+
+def _get_okta_groups(api_client):
+    """
+    Get groups from Okta server
+    :param api_client: Okta api client
+    :return: Array of Dictionary containing group properties
+    """
+    group_list = []
+    next_url = None
+
+    # SDK Bug
+    # get_paged_groups returns User object instead of UserGroup
+
+    while True:
+        try:
+            # https://developer.okta.com/docs/reference/api/groups/#list-groups
+            if next_url:
+                paged_response = api_client.get(next_url)
+            else:
+                params = {
+                    'limit': 10000,
+                }
+                paged_response = api_client.get_path('/', params)
+        except OktaError as okta_error:
+            logger.debug(f"Got error while going through list group {okta_error}")
```

Comment:
Why is this a debug level log message? Doesn't this indicate a complete failure to sync Okta?